### PR TITLE
Copy `cortex/pkg/chunk/purger` package dependency into Loki

### PR DESCRIPTION
**Pull request by DylanGuedes**
**What this PR does / why we need it**:
As a part of removing Loki dependency on cortex packages, we decided to copy the `chunk/purger` package from cortex to Loki itself.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Review by sandeepsukhani — COMMENTED** [review]
We do not use purger from cortex since we have our own implementation of handling deletion in the compactor. You can drop all the usages of it. Please feel free to ping me if you need any help.

**Comment by DylanGuedes** [comment]
> We do not use purger from cortex since we have our own implementation of handling deletion in the compactor. You can drop all the usages of it. Please feel free to ping me if you need any help.

Oops, good to know, thank you. I'll migrate to it then.

**Comment by DylanGuedes** [comment]
@sandeepsukhani  I pushed a commit that completely get rid of `chunk/purger` as you commented about Loki only relying on the Compactor deletion. WDYT?

File: pkg/querier/base/querier.go [hunk 1]
```diff
@@ -144,7 +142,7 @@ func NewChunkStoreQueryable(cfg Config, chunkStore chunkstore.ChunkStore) storag
 }
 
 // New builds a queryable and promql engine.
-func New(cfg Config, limits *validation.Overrides, distributor Distributor, stores []QueryableWithFilter, tombstonesLoader *purger.TombstonesLoader, reg prometheus.Registerer, logger log.Logger) (storage.SampleAndChunkQueryable, storage.ExemplarQueryable, *promql.Engine) {
+func New(cfg Config, limits *validation.Overrides, distributor Distributor, stores []QueryableWithFilter, reg prometheus.Registerer, logger log.Logger) (storage.SampleAndChunkQueryable, storage.ExemplarQueryable, *promql.Engine) {
```
Comment (cyriltovena) [inline pkg/querier/base/querier.go:145]: Who needs this ?

Comment (DylanGuedes) [reply pkg/querier/base/querier.go:145]: Some tests under `ruler/base`. And `ruler/base` is used by our own Ruler.

**Review by sandeepsukhani — APPROVED** [review]
LGTM

**Review by kavirajk — APPROVED** [review]
LGTM!

**Pull request merged**